src/units/abilities.cpp

```diff
@@ -867,41 +867,118 @@ static void add_name(std::string& weapon_abilities, bool active, const config::a
 std::string attack_type::weapon_specials(bool only_active, bool is_backstab) const
 {
 	//log_scope("weapon_specials");
-	std::string res;
-	for (const config::any_child sp : specials_.all_children_range())
-	{
-		const bool active = special_active(sp.cfg, AFFECT_EITHER, sp.key, is_backstab);
+	return weapon_specials_impl(only_active, is_backstab);
+}
 
-		const std::string& name =
-			active
-			? sp.cfg["name"].str()
-			: default_value(sp.cfg, "name_inactive", "name").str();
-		if (!name.empty()) {
-			if (!res.empty()) res += ", ";
-			if (only_active && !active) res += font::span_color(font::inactive_details_color);
-			res += name;
-			if (only_active && !active) res += "</span>";
+std::string attack_type::weapon_specials_value(const std::set<std::string> checking_tags, bool only_active, bool is_backstab) const
+{
+	//log_scope("weapon_specials_value");
+	return weapon_specials_impl(only_active, is_backstab, checking_tags, true);
+}
+
+std::string attack_type::weapon_specials_impl(bool only_active, bool is_backstab, const std::set<std::string> checking_tags, bool is_tooltip) const
+{
+	std::string res;
+	for (const config::any_child sp : specials_.all_children_range()) {
+		bool tag_checked = is_tooltip ? (checking_tags.count(sp.key) != 0) : true;
+		if(tag_checked){
```

Comment:
Throughout this function, `is_tooltip` seems to be used for multiple things that aren't logically connected to each other.

In this line and the line above it, it seems to be relying on the caller to pass `is_tooltip==true` if and only if it wants to do filtering using the `checking_tags`. If I'm understanding that right, I'd replace these two lines with
```
		if(checking_tags.empty() || checking_tags.count(sp.key) != 0) {
```

Reply:
here is_tooltip define if all tags can be used or only one affected to a stats("chance_to_hit" for chance to hit stats) if is_toolitip is true, then only these tags can be used, else, if all tags.